diffhtwo/experimental/tests/test_diffstarpop_halpha_opt.py

```diff
@@ -0,0 +1,248 @@
+import numpy as np
+import jax.numpy as jnp
+from jax import random as jran
+from dsps.cosmology import DEFAULT_COSMOLOGY
+from dsps.data_loaders import retrieve_fake_fsps_data
+from jax.flatten_util import ravel_pytree
+from numbers import Real
+from diffstar.defaults import T_TABLE_MIN
+from diffstarpop.defaults import DEFAULT_DIFFSTARPOP_PARAMS
+from dsps.cosmology import flat_wcdm
+from dsps.metallicity import umzr
+from diffsky.experimental import mc_lightcone_halos as mclh
+from diffsky.param_utils import spspop_param_utils as spspu
+from ..diffstarpop_halpha import diffstarpop_halpha_kern as dpop_halpha_L
+from ..diffstarpop_halpha import (
+    diffstarpop_halpha_lf_weighted as dpop_halpha_lf_weighted,
+)
+from ..diffstarpop_halpha_opt import fit_diffstarpop
+
+# from diffsky.experimental.scatter import DEFAULT_SCATTER_PARAMS
+# from diffsky.ssp_err_model import ssp_err_model
+
+
+theta_default, unravel_fn = ravel_pytree(DEFAULT_DIFFSTARPOP_PARAMS)
+idx = jnp.array([1, 8, 16])
+
+
+def _jitter_value(v, s, rng):
+    if isinstance(v, Real):
+        eps = s * (abs(v) if v != 0 else 1.0)
+        return v + rng.normal(0.0, eps)
+    return v
+
+
+def perturb_namedtuple(nt, scale=0.01, rng=None, bounds=None, per_field_scale=None):
```

Comment:
I guess this function came from the internet? Or maybe a bot? I think I see what it's doing. But now that you have a handy one-liner to create a flattened array from a nested namedtuple, is this function actually necessary?